shared/elements/gaia_grid/js/grid_view.js

```diff
@@ -467,11 +469,15 @@
           var lastItemInRow = this.items[idx - 1];
           step(lastItemInRow);
         }
-
+        var reversedPositions =
+              (windowWidth - this.layout.gridItemWidth) - (x * this.layout.gridItemWidth);
+        var decideXPosition = isRTL ?
+                  reversedPositions : x * this.layout.gridItemWidth;
```

Comment:
This should be in the same block as setCoordinates, otherwise we're doing extra work for nothing. I also think it would be better expressed like this:

var xPosition = x * this.layout.gridItemWidth;
if (isRTL) {
  xPosition = (windowWidth - this.layout.gridItemWidth) - xPosition;
}